ca/certificates.go

```diff
@@ -381,9 +387,24 @@ func NewRootCA(certBytes, keyBytes []byte, certExpiry time.Duration) (RootCA, er
 	// Calculate the digest for our Root CA bundle
 	digest := digest.FromBytes(certBytes)
 
+	// TODO(cyli): We currently do not yet support arbitrary chains of intermediates (e.g. the case of an offline root, with the CA signing with
+	// an intermediate) - we currently only only intermediates for which the first intermediate is cross-signed version of the signing (first) root
+	// for the purposes of root rotation.  If we wanted to support offline roots, rather than require at the first intermediate is the cross-signed
+	// version of the signing CA, it should *either* be a cross-signed version of the signing CA or it should be the issuer of the signing CA.
```

Comment:
Oh I see what you mean.  Ok, how I was interpreting this to work was:

Right now, we require that the first cert of the intermediate be a cross-signed root.  But that shouldn't be a requirement if we support the offline root case.  I originally thought for the offline root case, we'd append our intermediate signing CA, and then append these intermediates, which would chain up to the root.

But you're right that the intermediate signing CA can then just be included in the list of intermediates in which case it will still have the same public key and subject as the current signing CA, and that would make this logic simpler.  I'll update.